scripts/add-cluster.sh

```diff
@@ -282,16 +282,29 @@ else
 fi
 
 echo "Getting ${JOINING_CLUSTER_TYPE} SA token"
-SA_SECRET=`oc get sa ${SA_NAME} -n ${OPERATOR_NS} -o json ${OC_ADDITIONAL_PARAMS} | jq -r .secrets[].name | grep token`
-echo "SA secret found: ${SA_SECRET}"
-SA_TOKEN=`oc get secret ${SA_SECRET} -n ${OPERATOR_NS}  -o json ${OC_ADDITIONAL_PARAMS} | jq -r '.data["token"]' | base64 --decode`
+SERVER_VERSION=$(kubectl version -o json ${OC_ADDITIONAL_PARAMS} | jq -r .serverVersion.minor)
+if [[ ${SERVER_VERSION} -gt 23 ]]; then
+  echo "Running kube 1.24 or newer"
+  CLIENT_VERSION=$(kubectl version -o json ${OC_ADDITIONAL_PARAMS} | jq -r .clientVersion.minor)
+  if [[ ${CLIENT_VERSION} -lt 24 ]]; then
+    echo "ERROR: Since the cluster is running on Kubernetes 1.24 or newer, then you need to update your kubectl to match the same version."
+    echo "ERROR: Your kubectl version is $(kubectl version -o json ${OC_ADDITIONAL_PARAMS} | jq -r .clientVersion.gitVersion), but should be 1.24 or newer"
+    exit 1
+  fi
+  # create a token with duration of 100 years
```

Comment:
this is optimistic :)

Reply:
let's hope that a Sandbox instance won't run longer without reconfiguration :-D